### PR TITLE
HOSS2-168-NotAva Refactor

### PR DESCRIPTION
**Pull request by AdamCoops**

### CODE REVIEW

File: app/uk/gov/hmrc/homeofficeimmigrationstatus/views/StatusNotAvailablePageContext.scala [hunk 1]
```diff
@@ -16,11 +16,20 @@
...
 
   def fullName: String = s"${query.givenName} ${query.familyName}"
 
+  def notAvailablePersonalData() =
+    Seq(
+      Some(Row("nino", "generic.nino", query.nino.formatted)),
+      Some(Row("givenName", "generic.givenName", query.givenName)),
+      Some(Row("familyName", "generic.familyName", query.familyName)),
+      Some(Row("dob", "generic.dob", query.dateOfBirth))
+    ).flatten
```
Comment (liampan) [inline app/uk/gov/hmrc/homeofficeimmigrationstatus/views/StatusNotAvailablePageContext.scala:34]: As these are all Some, we can loose them and drop the flatten, saving the compiler a job :)

File: app/uk/gov/hmrc/homeofficeimmigrationstatus/views/error_template.scala.html [hunk 2]
```diff
@@ -38,11 +38,9 @@ <h1 class="govuk-heading-xl" id="title">@messages(heading)</h1>
 
         <p class="govuk-body">
             @messages("status-check-failure.contact-helpdesk-text")
-            <a href="@hd" class="govuk-link" target="_blank"> @messages("status-check-failure.contact-helpdesk-link")</a>
+            <a href="@hd" rel="noopener noreferrer" class="govuk-link" target="_blank"> @messages("status-check-failure.contact-helpdesk-link")</a>
```
Comment (liampan) [inline app/uk/gov/hmrc/homeofficeimmigrationstatus/views/error_template.scala.html:41]: nice spot :+1:

File: app/uk/gov/hmrc/homeofficeimmigrationstatus/views/StatusNotAvailablePage.scala.html [hunk 3]
```diff
@@ -14,12 +14,15 @@
...
+@import uk.gov.hmrc.homeofficeimmigrationstatus.viewmodels.{RowViewModel => Row}
 
 @this(
     govukButton : GovukButton,
-    govukWrapper: govuk_wrapper
+    govukWrapper: govuk_wrapper,
+    summaryList: components.SummaryList,
 )
 
 @(context: StatusNotAvailablePageContext)(implicit request: Request[_], messages: Messages, appConfig: AppConfig)
```
Comment (liampan) [inline app/uk/gov/hmrc/homeofficeimmigrationstatus/views/StatusNotAvailablePage.scala.html:28]: I can't add a comment there, but i think `@locale` can be removed now :)

File: app/uk/gov/hmrc/homeofficeimmigrationstatus/views/StatusNotAvailablePage.scala.html [hunk 4]
```diff
@@ -14,12 +14,15 @@
  * limitations under the License.
  *@
 
-@import uk.gov.hmrc.homeofficeimmigrationstatus.views.{DateFormat, StatusNotAvailablePageContext}
+@import uk.gov.hmrc.homeofficeimmigrationstatus.views.{DateFormat, ISO31661Alpha3, StatusNotAvailablePageContext}
```
Comment (liampan) [inline app/uk/gov/hmrc/homeofficeimmigrationstatus/views/StatusNotAvailablePage.scala.html:17]: why is this imported?

File: it/uk/gov/hmrc/homeofficeimmigrationstatus/controllers/HomeOfficeImmigrationStatusFrontendControllerISpec.scala [hunk 5]
```diff
@@ -210,7 +210,6 @@ class HomeOfficeImmigrationStatusFrontendControllerISpec
         checkHtmlResultWithBodyText(result, htmlEscapedMessage("status-not-available.title"))
         checkHtmlResultWithBodyText(result, query.nino.formatted)
         checkHtmlResultWithBodyText(result, s"${query.givenName} ${query.familyName}")
-        checkHtmlResultWithBodyText(result, "31 January 2001")
```
Comment (liampan) [inline it/uk/gov/hmrc/homeofficeimmigrationstatus/controllers/HomeOfficeImmigrationStatusFrontendControllerISpec.scala:213]: This test should still be passing. I think we need a spec for the view, The formatting for DOB is wrong

**Review by liampan — CHANGES_REQUESTED** [review]

File: it/uk/gov/hmrc/homeofficeimmigrationstatus/controllers/HomeOfficeImmigrationStatusFrontendControllerISpec.scala (hunk 5 above)
Comment (AdamCoops) [reply it/uk/gov/hmrc/homeofficeimmigrationstatus/controllers/HomeOfficeImmigrationStatusFrontendControllerISpec.scala:213]: Yeah, so this test is broke and causes the failure. As to why or how, not sure yet. the DOB format exists in two forms in that view-spec, and all of those pass so unless we change how we display the date? or the data has changed for this test maybe.

File: test/uk/gov/hmrc/homeofficeimmigrationstatus/views/StatusNotAvailableViewSpec.scala [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+    }
+
+    "have a status list content" in {
+      assertElementHasText(
+        doc,
+        "#not-available-list",
+        messages("status-not-available.list-item1") + " "
+          + messages("status-not-available.list-item2"))
+    }
+  }
```
Comment (liampan) [inline test/uk/gov/hmrc/homeofficeimmigrationstatus/views/StatusNotAvailableViewSpec.scala:69]: just need a test for the summary list to be rendered

```suggestion
     "have the summary list" in {
         assertRenderedById("notAvailablePersonalData")
     }
  }
```

File: app/uk/gov/hmrc/homeofficeimmigrationstatus/views/StatusNotAvailablePageContext.scala [hunk 7]
```diff
@@ -16,11 +16,20 @@
...
+import play.api.i18n.Messages
 import play.api.mvc.Call
 import uk.gov.hmrc.homeofficeimmigrationstatus.models.StatusCheckByNinoRequest
+import uk.gov.hmrc.homeofficeimmigrationstatus.viewmodels.{RowViewModel => Row}
 
 case class StatusNotAvailablePageContext(query: StatusCheckByNinoRequest, searchAgainCall: Call) {
 
   def fullName: String = s"${query.givenName} ${query.familyName}"
 
+  def notAvailablePersonalData(implicit messages: Messages) =
```
Comment (liampan) [inline app/uk/gov/hmrc/homeofficeimmigrationstatus/views/StatusNotAvailablePageContext.scala:28]: a test will needed added to the contextSpec for the construction of this Sequence

File: test/uk/gov/hmrc/homeofficeimmigrationstatus/views/StatusNotAvailableViewSpec.scala (hunk 6 above)
Comment (liampan) [reply test/uk/gov/hmrc/homeofficeimmigrationstatus/views/StatusNotAvailableViewSpec.scala:69]: and a test to assert the button is present can use a verify for this as you've mocked it if you use the searchAgainButton Component

**Pull request merged**